### ISSUE TITLE
Make it so tests don't randomly fail ALL THE TIME

### ISSUE THREAD

**Issue by mpharrigan** [msmbuilder/msmbuilder#792]
cc #758 

Every time I see a failure, I'm going to copy the error here so we know what tests are the problem children

```
======================================================================
FAIL: msmbuilder.tests.test_lumping.test_ntimescales_1
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Miniconda3-x64\envs\_test\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "c:\miniconda3-x64\envs\_test\lib\site-packages\msmbuilder-3.5.0.dev0-py3.5-win-amd64.egg\msmbuilder\tests\test_lumping.py", line 81, in test_ntimescales_1
    assert len(np.unique(lumped_trajs)) == 11
AssertionError: 
-------------------- >> begin captured stdout << ---------------------
MSM contains 1 strongly connected component above weight=1.00. Component 0 selected, with population 100.000000%
Warning: Maximum number of function evaluations has been exceeded.
 
-----
```

**Comment by mpharrigan** [comment]
```
Error: HTTPError: 500 Server Error: INTERNAL SERVER ERROR for url: https://conda.anaconda.org/omnia/win-32/mdtraj-1.6.1-np110py35_0.tar.bz2: https://conda.anaconda.org/omnia/win-32/mdtraj-1.6.1-np110py35_0.tar.bz2
 
Command exited with code 1
```

**Comment by cxhernandez** [comment]
in [appveyor](https://ci.appveyor.com/project/MatthewHarrigan/msmbuilder/build/1.0.439/job/3ixqe3btf5gqc1ya#L1302)
```
======================================================================
ERROR: msmbuilder.tests.test_msm_uncertainty.test_doublewell
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Miniconda3\envs\_test\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "c:\miniconda3\envs\_test\lib\site-packages\msmbuilder-3.5.0.dev0-py3.5-win32.egg\msmbuilder\tests\test_msm_uncertainty.py", line 60, in test_doublewell
    print(model2.uncertainty_timescales())
  File "c:\miniconda3\envs\_test\lib\site-packages\msmbuilder-3.5.0.dev0-py3.5-win32.egg\msmbuilder\msm\ratematrix.py", line 257, in uncertainty_timescales
    self._build_information()
  File "c:\miniconda3\envs\_test\lib\site-packages\msmbuilder-3.5.0.dev0-py3.5-win32.egg\msmbuilder\msm\ratematrix.py", line 303, in _build_information
    self.information_[np.ix_(inds, inds)] = scipy.linalg.pinv(-hessian)
  File "C:\Miniconda3\envs\_test\lib\site-packages\scipy\linalg\basic.py", line 972, in pinv
    a = _asarray_validated(a, check_finite=check_finite)
  File "C:\Miniconda3\envs\_test\lib\site-packages\scipy\_lib\_util.py", line 187, in _asarray_validated
    a = toarray(a)
  File "C:\Miniconda3\envs\_test\lib\site-packages\numpy\lib\function_base.py", line 668, in asarray_chkfinite
    "array must not contain infs or NaNs")
ValueError: array must not contain infs or NaNs
```

**Comment by mpharrigan** [comment]
```
ERROR: msmbuilder.tests.test_apm.test_shapes

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/miniconda3/envs/_test/lib/python3.4/site-packages/nose/case.py", line 198, in runTest

    self.test(*self.arg)

  File "/home/travis/miniconda3/envs/_test/lib/python3.4/site-packages/msmbuilder-3.5.0.dev0-py3.4-linux-x86_64.egg/msmbuilder/tests/test_apm.py", line 17, in test_shapes

    m.fit([np.random.randn(100, 2)])

  File "/home/travis/miniconda3/envs/_test/lib/python3.4/site-packages/msmbuilder-3.5.0.dev0-py3.4-linux-x86_64.egg/msmbuilder/cluster/apm.py", line 94, in fit

    self._run()

  File "/home/travis/miniconda3/envs/_test/lib/python3.4/site-packages/msmbuilder-3.5.0.dev0-py3.4-linux-x86_64.egg/msmbuilder/cluster/apm.py", line 118, in _run

    self._do_time_clustering(macro_state=k)

  File "/home/travis/miniconda3/envs/_test/lib/python3.4/site-packages/msmbuilder-3.5.0.dev0-py3.4-linux-x86_64.egg/msmbuilder/cluster/apm.py", line 208, in _do_time_clustering

    if self._get_Lagtime(micro_state, macro_state) is 0:

  File "/home/travis/miniconda3/envs/_test/lib/python3.4/site-packages/msmbuilder-3.5.0.dev0-py3.4-linux-x86_64.egg/msmbuilder/cluster/apm.py", line 170, in _get_Lagtime

    if self._get_RelaxProb(micro_state, macro_state) > 0.6321206:

  File "/home/travis/miniconda3/envs/_test/lib/python3.4/site-packages/msmbuilder-3.5.0.dev0-py3.4-linux-x86_64.egg/msmbuilder/cluster/apm.py", line 189, in _get_RelaxProb

    self.lag_time] != micro_state or self.__temp_MacroAssignments_[

nose.proxy.IndexError: index 98 is out of bounds for axis 0 with size 98
```

**Comment by cxhernandez** [comment]
```
======================================================================
FAIL: msmbuilder.tests.test_kmedoids.test_multitraj_cluster_ids_1
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda3/envs/_test/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/travis/miniconda3/envs/_test/lib/python3.5/site-packages/msmbuilder-3.5.0.dev0-py3.5-linux-x86_64.egg/msmbuilder/tests/test_kmedoids.py", line 105, in test_multitraj_cluster_ids_1
    or (id1[0] == 1 and id2[0] == 0))
AssertionError
```

**Comment by mpharrigan** [comment]
The windows timeout I don't know how to fix other than wait for #805
